cachito/workers/pkg_managers/rubygems.py

```diff
@@ -312,11 +322,114 @@ def _download_git_package(gem, rubygems_deps_dir, rubygems_raw_repo_name, nexus_
         "raw_component_name": raw_component_name,
         "have_raw_component": have_raw_component,
     }
+
+
+def _get_path_package_info(dep, bundle_dir, package_root):
+    """
+    Get info about PATH dependency including path relative to the bundle source root.
+
+    :param GemMetadata dep: path dependency
+    :param bundle_dir: the root of the bundle with app source code
+    :param package_root: path to the root of the processed package
+    :return: dict with name, version and path keys
+    """
+    path = Path(package_root / dep.source).resolve().relative_to(bundle_dir.source_root_dir)
+
+    return {
+        "name": dep.name,
+        "version": dep.version,
+        "path": path,
+    }
+
+
+def resolve_rubygems(package_root, request):
+    """
+    Resolve and fetch RubyGems dependencies for the given app source archive.
+
+    :param Path package_root: the full path to the package root
+    :param dict request: the Cachito request to resolve RubyGems dependencies for
+    :return: a dictionary that has the following keys:
+        ``dependencies`` which is a list of dicts representing the package Dependencies
+        ``gemfile_lock`` an absolute path to the Gemfile.lock
+    """
+    gemlock_path = package_root / GEMFILE_LOCK
+    dependencies = parse_gemlock(package_root, gemlock_path)
+    dependencies = download_dependencies(request["id"], dependencies, package_root)
+
+    rubygems_repo_name = get_rubygems_hosted_name(request["id"])
+    for dependency in dependencies:
+        if dependency["kind"] == "GEM":
+            _push_downloaded_gem(dependency, rubygems_repo_name)
+
+    return {
+        "package": {"type": "rubygems"},
+        "dependencies": dependencies,
+        "gemfile_lock": gemlock_path,
+    }
+
+
+def _upload_rubygems_package(repo_name, artifact_path):
+    """
+    Upload a RubyGems package to a Nexus repository.
+
+    :param str repo_name: the name of the hosted RubyGems repository to upload the package to
+    :param str artifact_path: the path for the RubyGems package to be uploaded
+    """
+    log.debug(
+        "Uploading %r as a RubyGems package to the %r Nexus repository", artifact_path, repo_name
+    )
+    nexus.upload_asset_only_component(repo_name, "rubygems", artifact_path, to_nexus_hoster=False)
+
+
+def _push_downloaded_gem(dependency, rubygems_repo_name):
+    """
+    Upload a GEM dependency to the request temporary Nexus repository.
+
+    :param dict dependency: Single entry with the info about downloaded package retrieved from the
+        list returned by the download_dependencies function
+    :param str rubygems_repo_name: Name of the Nexus RubyGems hosted repository to push
+        the requirement to
+    :return: dict with the cachito Dependency representation
+    :rtype: dict
+    :raises CachitoError: If Nexus upload operation fails
+    """
+    try:
+        _upload_rubygems_package(rubygems_repo_name, dependency["path"])
+    except CachitoError:
```

Comment:
We recently changed error classification to be more detailed for debugging. Should we change it to [NetworkError](https://github.com/containerbuildsystem/cachito/blob/master/cachito/workers/nexus.py#L387)? https://github.com/containerbuildsystem/cachito/pull/680

Reply:
+1  feel free to add new more precise error type for your case. :)